tools/isobuild/meteor-npm.js

```diff
@@ -258,9 +258,10 @@ var updateExistingNpmDirectory = function (packageName, newPackageNpmDir,
     // `npm install`
     installFromShrinkwrap(newPackageNpmDir);
 
-    // delete package.json and npm-shrinkwrap.json
-    files.unlink(files.pathJoin(newPackageNpmDir, 'package.json'));
-    files.unlink(files.pathJoin(newPackageNpmDir, 'npm-shrinkwrap.json'));
+    // Back when we used npm@1.4.28, we deleted the package.json and
+    // npm-shrinkwrap.json files created above, but npm@3.3.9 needs them
+    // to remain undeleted to avoid those pesky "couldn't read npm version
+    // lock information" errors.
```

Comment:
After looking at this code some more, I think we can keep deleting those files as long as we do it after the call to `getInstalledDependencies` (which runs the failing `npm ls --json` command). I'll push another commit momentarily.